### DISCUSSION TITLE
How to limit fields returned by List api?

### DISCUSSION THREAD

**Discussion by luuhai48** [pocketbase/pocketbase#1880] [Q&A]
I just started using pocketbase for personal project. When calling List posts API, it returns every fields in the DB, including content of the post, which is long and not needed.
Is there a way I can select only fields that I need?
Thank you.

**Comment by baguetteism** [comment]


**Reply by luuhai48** [reply]
Hi @xinnypie 
I managed to make some changes, now the record list can have query param `?selects=email,name`, which returns only fields that I need, and others will be empty string.
Could it be contributed to the source code? (I know it missing a lot like changes in other APIs, tests...)
https://github.com/luuhai48/pocketbase/pull/1/commits/b322c29d926a6958afc24bef458d9d6801ec4daf

**Reply by ganigeorgiev** [reply]
@xinnypie, @luuhai48 Support for `?fields=field1,field2` query parameter to limit the returned API fields was added in the [`develop`](https://github.com/pocketbase/pocketbase/tree/develop) branch and will be available with the next PocketBase v0.16.0 release (there are no ETAs yet).
